deepchem/utils/fake_data_generator.py

```diff
@@ -0,0 +1,157 @@
+  2) The generated dataset in FakeDataset have same node feature distribution
+  irrespective of graph or node label which make it unsuitable to use during testing.
+  """
+
+  # TODO @arunppsg Once torch_geometric's version 2.0.4 is release which
+  # fixes above two issues, this can be depreciated.
+  def __init__(self,
+               avg_n_nodes: int = 10,
+               n_node_features: int = 5,
+               avg_degree: int = 4,
+               n_edge_features: int = 3,
+               n_classes: int = 2,
+               task: str = 'graph',
+               **kwargs):
+    """
+    Parameters
+    ----------
+    avg_n_nodes: int, default 10
+      Average number of nodes in a graph
+    n_node_features: int, default 5
+      Average number of node features in a graph
+    avg_degree: int, default 4
+      Average degree of the graph
+    n_edge_features: int, default 3
+      Average number of features in the edge
+    task: str, default 'graph'
+      Indicates node-level labels or graph-level labels
+    kwargs: optional
+      Additional graph attributes and their shapes , e.g. `global_features = 5`
+    """
+    self.avg_n_nodes = max(avg_n_nodes, avg_degree)
+    self.avg_degree = max(avg_degree, 1)
+    self.n_node_features = n_node_features
+    self.n_edge_features = n_edge_features
+    self.n_classes = n_classes
+    self.task = task
+    self.kwargs = kwargs
+
+  def sample(self, n_graphs: int = 100) -> NumpyDataset:
+    """Samples graphs
+
+    Parameters
+    ----------
+    n_graphs: int, default 100
+      Number of graphs to generate
+
+    Returns
+    -------
+    graphs: NumpyDataset
+      Generated Graphs
+    """
+    graphs, labels = [], []
+    for i in range(n_graphs):
+      if self.task == 'graph':
+        label = random.randint(0, self.n_classes - 1)
+      elif self.task == 'node':
+        # TODO
+        pass
+      n_nodes = get_num_nodes(self.avg_n_nodes, self.avg_degree)
+      edge_index = get_edge_index(n_nodes, self.avg_degree)
+      n_edges = edge_index.shape[1]
+      node_features = np.random.rand(n_nodes, self.n_node_features) + label
+      edge_features = np.random.rand(n_edges, self.n_edge_features) + label
+      kwargs = {}
+      for feature_name, feature_shape in self.kwargs.items():
+        kwargs[feature_name] = np.random.rand(1, feature_shape) + label
+      graph = GraphData(node_features, edge_index, edge_features, **kwargs)
+      graphs.append(graph)
+      labels.append(label)
+
+    graph_dataset = NumpyDataset(X=np.array(graphs), y=np.array(labels))
+    return graph_dataset
+
+
+def get_num_nodes(avg_n_nodes: int, avg_degree: int) -> int:
+  """Returns the number of nodes in a graph sampled uniformly with
+  with mean number of nodes as avg_n_nodes.
+
+  Parameters
+  ----------
+  avg_n_nodes: int
+    Average number of nodes in the graph
+  avg_degree: int
+    Average degree in a graph
+  """
+  min_num_nodes = max(3 * avg_n_nodes // 4, avg_degree)
```

Comment:
3/4/5 are magic numbers. Instead they should be variables with names and possibly even keyword arguments if it would make sense for users to be able to modify these defaults.

Reply:
Added min_nodes and max_nodes as user input arguments.